File: shared/src/test/scala/stasis/test/specs/unit/shared/interop/users/ResetUserPasswordInteropSpec.scala
package stasis.test.specs.unit.shared.interop.users

import stasis.shared.api.Formats._
import stasis.shared.api.requests.ResetUserPassword
import stasis.shared.api.responses.UpdatedUserSalt
import stasis.test.specs.unit.shared.interop.InteropSpec

class ResetUserPasswordInteropSpec extends InteropSpec {
  "ResetUserPassword interop" should "decode and re-encode a request" in {
    assert(
      domain = "users",
      resource = "ResetUserPassword",
      matches = ResetUserPassword(rawPassword = "new-password-value")
    )
  }

  it should "decode and re-encode UpdatedUserSalt" in {
    assert(
      domain = "users",
      resource = "UpdatedUserSalt",
      matches = UpdatedUserSalt(salt = "new-salt-value")
    )
  }
}

File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/interop/users/ResetUserPasswordInteropSpec.kt
package stasis.test.client_android.lib.interop.users

import stasis.client_android.lib.model.server.api.requests.ResetUserPassword
import stasis.client_android.lib.model.server.api.responses.UpdatedUserSalt
import stasis.test.client_android.lib.interop.InteropSpec

class ResetUserPasswordInteropSpec : InteropSpec({
    "ResetUserPassword interop" should {
        "decode and re-encode a request" {
            assert(
                domain = "users",
                resource = "ResetUserPassword",
                matches = ResetUserPassword(rawPassword = "new-password-value")
            )
        }

        "decode and re-encode UpdatedUserSalt" {
            assert(
                domain = "users",
                resource = "UpdatedUserSalt",
                matches = UpdatedUserSalt(salt = "new-salt-value")
            )
        }
    }
})
